docker/cassandra/default-config/cassandra-env.sh.d/002-cassandra-limits.sh

```diff
@@ -0,0 +1 @@
+ulimit -l unlimited
```

Comment:
Moved this from `ConfigMap.go` to here, based on the TODO.

`default-config/cassandra-env.sh` (aka, `/etc/cassandra/cassandra-env.sh` inside the built image) is blank.
Open for discussion as to whether this line (and similar ones) should go in a _fragment file_ (as this PR introduces) or into the "main" `cassandra-env.sh` file.

The intent of _fragment files_ is to allow for easy additions of new config settings, especially for optional features, by either the operator (e.g, if prometheus support is enabled, drop in a fragment that appends to `JVM_OPTS`) or by users (set some custom JVM params, alter the environment, etc).

Fragments are easy to add, but harder to remove -- the user would have to create a config map containing an empty file with the same name (i.e, `cassandra-env.sh.d/002-cassandra-limits.sh` in this case).

Reply:
specifically ulimits etc and other bits should be part of the privileged initi container? not part of the C* container

Reply:
In the Java code, inject a fragment into the Cassandra container, identical to this.
But you're right, we're not calling `sysctl` anywhere -- I missed the init container.
Should the ulimits call be in the init container too?

Reply:
See https://github.com/instaclustr/cassandra-operator/blob/master/java/operator/src/main/java/com/instaclustr/cassandra/operator/controller/DataCenterReconciliationController.java#L412

Reply:
this is what gets set within the priv container https://github.com/instaclustr/cassandra-operator/blob/8fed62125a785025fe3868ce579b809f9f07cc5e/java/operator/src/main/java/com/instaclustr/cassandra/operator/controller/DataCenterReconciliationController.java#L338-L345

then the C* container itself has the following kernel capabilities assigned:

https://github.com/instaclustr/cassandra-operator/blob/8fed62125a785025fe3868ce579b809f9f07cc5e/java/operator/src/main/java/com/instaclustr/cassandra/operator/controller/DataCenterReconciliationController.java#L150-L153

should the actual init container have those kernel capabilities instead and the C* container is less privileged?? I'm not sure, but that would feel to me like a good reduction of permissions...

Reply:
_dives deep into Kernel security contexts_

Reply:
C* definitely needs IPC_LOCK for itself, so can't move that
ulimit unlimited is the amount it can memlock, so again those two on the C* container are fine as its that process itself that needs those perms

SYS_RESOURCE  http://man7.org/linux/man-pages/man7/capabilities.7.html is file refs 

iirc the init container had to be done as that was a sysctl setting rather than a kernal capability on a process?

Reply:
Yeah, thinking about it more...

Pods are basically container groups. We need to run sysctl in a privileged container, because that's essentially modifying a host kernel parameter. It will persist even after the container/pod/etc has terminated.

`ulimits` is a bash built-in, and it modifies the limits to all subprocesses started inside that instance of the shell, but not other processes, and will be "reset" (more, forgotten) when the container terminates. Because we want the limits to apply specifically to Cassandra, we run it before we exec the JVM.

Reply:
ok makes sense

Reply:
I would go with fragments unless it is something absolutely crucial for whole container / pod to work correctly which would render it unusable.